### PR TITLE
feat: Add pom.xml support for sonarype SAST

### PR DESCRIPTION
**Pull request by laurentsimon**
See https://github.com/ossf/scorecard/issues/2085

/cc @edwardsph

### CODE REVIEW

File: checks/sast.go [hunk 1]
```diff
@@ -205,3 +222,106 @@ func codeQLInCheckDefinitions(c *checker.CheckRequest) (int, error) {
...
+			Type:      checker.FileTypeSource,
+			Offset:    offset,
+			EndOffset: endOffset,
+		},
+	})
+
+	return true, nil
+}
+
+func findLine(content, data []byte) (uint, error) {
```
Comment (azeemshaikh38) [inline checks/sast.go:309]: What if there are multiple matching lines?

**Review by azeemshaikh38 — APPROVED** [review]
LGTM with comments.

File: checks/sast.go [hunk 2]
```diff
@@ -205,3 +222,106 @@ func codeQLInCheckDefinitions(c *checker.CheckRequest) (int, error) {
...
+	content []byte,
+	args ...interface{},
+) (bool, error) {
+	if path.Base(pathfn) != "pom.xml" {
+		return true, nil
+	}
+
+	if len(args) != 1 {
+		return false, fmt.Errorf(
+			"validateSonarConfig requires exactly 1 arguments: %w", errInvalid)
```
Comment (aidenwang9867) [inline checks/sast.go:269]: nit: argument

File: checks/sast.go [hunk 3]
```diff
@@ -205,3 +222,106 @@ func codeQLInCheckDefinitions(c *checker.CheckRequest) (int, error) {
...
+	}
+
+	// Verify the type of the data.
+	pdata, ok := args[0].(*[]sonarConfig)
+	if !ok {
+		return false, fmt.Errorf(
+			"validateSonarConfig expects arg[0] of type *[]sonarConfig]: %w", errInvalid)
+	}
+
+	regex := regexp.MustCompile(`<sonar\.host\.url>\s*(https:\/\/.*)\s*<\/sonar\.host\.url>`)
```
[the remark this answers is not in the document]
Comment (aidenwang9867) [reply checks/sast.go:279]: Found the default value of the host.url starts with `http://`:
https://docs.sonarqube.org/latest/analysis/analysis-parameters/

so I agree the regex could be more general ;)

File: checks/sast.go (hunk 2 above)
Comment (laurentsimon) [reply checks/sast.go:269]: good catch

File: checks/sast.go (hunk 1 above)
Comment (laurentsimon) [reply checks/sast.go:309]: For now it does not matter so I kept it simple. We get the first it's probably enough. I don't know if it's allowed to have multiple ones.

File: checks/sast.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (laurentsimon) [reply checks/sast.go:279]: I forgot to ignore https. That was in the back of my mind.. and I forgot :-)
Thanks

File: checks/sast.go [hunk 4]
```diff
@@ -205,3 +222,106 @@ func codeQLInCheckDefinitions(c *checker.CheckRequest) (int, error) {
...
+
+	return checker.MinResultScore, nil
+}
+
+// Check file content.
+var validateSonarConfig fileparser.DoWhileTrueOnFileContent = func(pathfn string,
+	content []byte,
+	args ...interface{},
+) (bool, error) {
+	if path.Base(pathfn) != "pom.xml" {
```
[the remark this answers is not in the document]
Comment (laurentsimon) [reply checks/sast.go:263]: can do that. I thought it was overkill, but it
s cheap. Will do.

**Comment by edwardsph** [comment]
Thanks for the quick change - looking forward to testing in the next release

**Pull request merged**